### PR TITLE
Remove get_var('PUBLIC_CLOUD') in faver of version_utils::is_public_cloud()

### PR DESCRIPTION
**Pull request by pdostal**

### CODE REVIEW

File: tests/sles4sap/saptune/mr_test_run.pm [hunk 1]
```diff
@@ -37,7 +37,7 @@ our $result_module;
 sub reboot_wait {
     my ($self) = @_;
 
-    if (is_public_cloud) {
+    if (is_public_cloud()) {
```
Comment (asmorodskyi) [inline tests/sles4sap/saptune/mr_test_run.pm:40]: thanks !

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by asmorodskyi** [comment]
A lot of places was a big surprise for me . What `is_public_cloud` is even doing there . But it is not related to this PR . Just a note for myself that I would like to investigate this PR later ...

**Review by asmorodskyi — APPROVED** [review]
LGTM 

P.S. and special thank you for using `()` !!! :D

**Comment by asmorodskyi** [comment]
I may ignore ` trigger_and_monitor_openqa` but not "Running unit tests with perl v5.32 (pull_request) "  can you please figure out what is going on https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/8733773597/job/23963113904?pr=18793

**Comment by pdostal** [comment]
> I may ignore  trigger_and_monitor_openqa but not "Running unit tests with perl v5.32 (pull_request) " can you please figure out what is going on https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/8733773597/job/23963113904?pr=18793
I'm clueless 😞 I asked @mpagot.

**Comment by mpagot** [comment]
> > I may ignore  trigger_and_monitor_openqa but not "Running unit tests with perl v5.32 (pull_request) " can you please figure out what is going on https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/8733773597/job/23963113904?pr=18793
> 
> I'm clueless 😞 I asked @mpagot.

Issue is not only in Unit Testing: look at https://github.com/os-autoinst/os-autoinst-distri-opensuse/actions/runs/8733773376/job/23963113237?pr=18793#step:5:14  -->  https://openqa.opensuse.org/tests/4091612

```
[2024-04-18T07:06:24.327234Z] [warn] [pid:86367] !!! main: "is_public_cloud" is not exported by the version_utils module
  Can't continue after import errors at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/lib/Utils/Backends.pm line 21.
  BEGIN failed--compilation aborted at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/lib/Utils/Backends.pm line 21.
  Compilation failed in require at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/lib/version_utils.pm line 13.
  BEGIN failed--compilation aborted at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/lib/version_utils.pm line 13.
  Compilation failed in require at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/products/opensuse/main.pm line 12.
  BEGIN failed--compilation aborted at /var/lib/openqa/pool/16/os-autoinst-distri-opensuse/products/opensuse/main.pm line 12.
  Compilation failed in require at /usr/lib/os-autoinst/OpenQA/Isotovideo/Utils.pm line 390.
```

**Pull request merged**